crates/k8s-version/src/group.rs

```diff
@@ -0,0 +1,66 @@
+use std::{fmt, ops::Deref, str::FromStr};
+
+use lazy_static::lazy_static;
+use regex::Regex;
+use snafu::{ensure, Snafu};
+
+const MAX_GROUP_LENGTH: usize = 253;
+
+lazy_static! {
+    static ref API_VERSION_REGEX: Regex =
+        Regex::new(r"^(?:(?:[a-z0-9][a-z0-9-]{0,61}[a-z0-9])\.?)+$").unwrap();
```

Comment:
I mostly use non-capturing groups when I don't care about individual parts (which is the case here). The regex is only in place to validate that the input follows the expected format (DNS subdomain). I'm not sure if there is any performance benefit when using them.

After reading this, I think the `API_VERSION_REGEX` static should be named `API_GROUP_REGEX`, because that's what we validate with it.

Reply:
Renamed in 8a38f47